bft/dialects/parser.py

```diff
@@ -1,43 +1,36 @@
 from bft.core.yaml_parser import BaseYamlParser, BaseYamlVisitor
-from bft.dialects.types import DialectFile, DialectFunction, DialectKernel
+from bft.dialects.types import DialectFile, DialectFunction, DialectKernel, short_type_to_type
 
 
 class DialectFileVisitor(BaseYamlVisitor[DialectFile]):
-    def visit_kernel(self, kernel):
-        arg_types = self._get_or_die(kernel, "args")
-        result_type = self._get_or_die(kernel, "result")
-        return DialectKernel(arg_types, result_type)
+    @staticmethod
+    def visit_kernel(kernel):
+        arg_types = []
+        if kernel != '':
+            arg_types = [DialectFileVisitor.get_long_type(arg_type) for arg_type in kernel.split("_")]
+        return DialectKernel(arg_types, any)
+
+    @staticmethod
+    def get_long_type(short_type):
+        long_type = short_type_to_type.get(short_type, None)
+        if long_type is None:
+            return short_type
+        return long_type
+
+    @staticmethod
+    def _get_unqualified_func_name(name):
+        return name.split(".")[-1]
 
     def visit_scalar_function(self, func):
```

Comment:
Looks like both `visit_scalar_function` and `visit_aggregate_function` just return `self._visit_function(func)` now.